### PR TITLE
Add Pose2 Component Jacobians

### PR DESCRIPTION
**Pull request by DanMcGann**
## Motivation
Pose3 component extraction functions (`rotation` and `translation`) optionally allow for jacobians. However, Pose2 does not parallel this interface. This can cause issues when defining code templated to handle either dimension of pose.

## Implementation
This PR updates the Pose2 interface to parallel the Pose3 interface by adding `OptionalJacobians` to Pose2's `translation` and `rotation` functions. 

The derivation of these Jacobians directly parallels that for Pose3, although we do have to account for the difference in tangent space parameter order between Pose2 and Pose3. 

Jacobians are verified with comparison to numerical Jacobian in added tests.

The python interface is also extended to support these changes.

## Effects
While these are interface level changes, they should not cause any backwards compatibility issues, as they are implemented with `OptionalJacobians` that default to `none`. Existing code should be agnostic to these changes.

### CODE REVIEW

File: gtsam/geometry/Pose2.h [hunk 1]
```diff
@@ -252,16 +252,16 @@ class Pose2: public LieGroup<Pose2, 3> {
   inline double theta() const { return r_.theta(); }
 
   /// translation
-  inline const Point2& t() const { return t_; }
+  inline const Point2& t() const { return translation(); }
```
[the remark this answers is not in the document]
Comment (DanMcGann) [reply gtsam/geometry/Pose2.h:255]: Agreed, at the moment `t()` and `r()` are effectively macros to `translation` and `rotation` respectively. Regarding the design decision I can see an argument on each side of making these inline. I am happy to update to the preference of the maintainers!

The parallel Pose3 functions are note inline if that provides signal to the decision process.

[the remark this answers is not in the document]
Comment (ProfFan) [reply gtsam/geometry/Pose2.h:255]: yeah make them inline :)

[the remark this answers is not in the document]
Comment (dellaert) [reply gtsam/geometry/Pose2.h:255]: Seems that this orthogonal to the aim of this PR. What's the problem with leaving these methods as is?

**Review by dellaert — COMMENTED** [review]
Very cool. If we can revert those "macro" methods (if you agree) then I'll merge. Happy to discuss more.

File: gtsam/geometry/Pose2.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanMcGann) [reply gtsam/geometry/Pose2.h]: No problem, just my own over-dedication to the one source of truth principle. Reverted!

**Review by dellaert — APPROVED** [review]

**Comment by dellaert** [comment]
Merge in develop to re-run windows CI?

**Pull request merged**